### PR TITLE
[Enhancement] Add [[nodiscard]] to functions

### PR DESCRIPTION
**Pull request by satanson**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by stdpain — DISMISSED** [review]

File: be/src/exec/hash_join_components.cpp [hunk 1]
```diff
@@ -70,15 +71,15 @@ void HashJoinBuilder::reset(const HashTableParam& param) {
 
 void HashJoinBuilder::reset_probe(RuntimeState* state) {
     _key_columns.clear();
-    _ht.reset_probe_state(state);
+    (void)_ht.reset_probe_state(state);
```
Comment (silverbullet233) [inline be/src/exec/hash_join_components.cpp:74]: how about change the return type of JoinHashTable::reset_probe_state to void directly ? it always return OK now.

File: be/src/exec/pipeline/runtime_filter_types.h [hunk 2]
```diff
@@ -220,14 +220,16 @@ class PartialRuntimeFilterMerger {
     // mark runtime_filter as always true.
     bool set_always_true() {
         _always_true = true;
-        return _try_do_merge({});
+        (void)_try_do_merge({});
```
Comment (silverbullet233) [inline be/src/exec/pipeline/runtime_filter_types.h:223]: is it safe to ignore error status returned by _try_to_merge?

**Comment by silverbullet233** [comment]
fixed #27471

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 3 / 21 (14.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hash_joiner.cpp|0|2|00.00%|[245, 549]|
|:large_blue_circle:|src/exec/hash_join_components.cpp|0|2|00.00%|[22, 26]|
|:large_blue_circle:|src/exec/hash_join_node.cpp|0|1|00.00%|[892]|
|:large_blue_circle:|src/exec/hash_joiner.h|0|6|00.00%|[263, 269, 272, 297, 314, 315]|
|:large_blue_circle:|src/exec/pipeline/hashjoin/hash_join_probe_operator.cpp|0|1|00.00%|[78]|
|:large_blue_circle:|src/exec/pipeline/runtime_filter_types.h|1|7|14.29%|[223, 231, 252, 310, 389, 397]|
|:large_blue_circle:|src/exec/query_cache/cache_manager.cpp|2|2|100.00%|[]|

File: be/src/exec/pipeline/runtime_filter_types.h (hunk 2 above)
Comment (satanson) [reply be/src/exec/pipeline/runtime_filter_types.h:223]: yes

**Review by silverbullet233 — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 3 / 21 (14.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hash_joiner.cpp|0|2|00.00%|[245, 549]|
|:large_blue_circle:|src/exec/hash_join_components.cpp|0|2|00.00%|[22, 26]|
|:large_blue_circle:|src/exec/hash_join_node.cpp|0|1|00.00%|[892]|
|:large_blue_circle:|src/exec/hash_joiner.h|0|6|00.00%|[263, 269, 272, 297, 314, 315]|
|:large_blue_circle:|src/exec/pipeline/hashjoin/hash_join_probe_operator.cpp|0|1|00.00%|[78]|
|:large_blue_circle:|src/exec/pipeline/runtime_filter_types.h|1|7|14.29%|[223, 231, 252, 310, 389, 397]|
|:large_blue_circle:|src/exec/query_cache/cache_manager.cpp|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 3 / 21 (14.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hash_joiner.cpp|0|2|00.00%|[245, 549]|
|:large_blue_circle:|src/exec/hash_join_components.cpp|0|2|00.00%|[22, 26]|
|:large_blue_circle:|src/exec/hash_join_node.cpp|0|1|00.00%|[892]|
|:large_blue_circle:|src/exec/hash_joiner.h|0|6|00.00%|[263, 269, 272, 297, 314, 315]|
|:large_blue_circle:|src/exec/pipeline/hashjoin/hash_join_probe_operator.cpp|0|1|00.00%|[78]|
|:large_blue_circle:|src/exec/pipeline/runtime_filter_types.h|1|7|14.29%|[223, 231, 252, 310, 389, 397]|
|:large_blue_circle:|src/exec/query_cache/cache_manager.cpp|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 3 / 22 (13.64%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hash_joiner.cpp|0|2|00.00%|[245, 549]|
|:large_blue_circle:|src/exec/hash_join_components.cpp|0|2|00.00%|[22, 26]|
|:large_blue_circle:|src/exec/hash_join_node.cpp|0|2|00.00%|[561, 892]|
|:large_blue_circle:|src/exec/hash_joiner.h|0|6|00.00%|[263, 269, 272, 297, 314, 315]|
|:large_blue_circle:|src/exec/pipeline/hashjoin/hash_join_probe_operator.cpp|0|1|00.00%|[78]|
|:large_blue_circle:|src/exec/pipeline/runtime_filter_types.h|1|7|14.29%|[223, 231, 252, 310, 389, 397]|
|:large_blue_circle:|src/exec/query_cache/cache_manager.cpp|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by silverbullet233 — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**